lib/EnvironmentPlugin.js

```diff
@@ -1,28 +1,49 @@
 /*
 	MIT License http://www.opensource.org/licenses/mit-license.php
-	Author Simen Brekken @simenbrekken
+	Authors Simen Brekken @simenbrekken, Einar Löve @einarlove
 */
 var DefinePlugin = require("./DefinePlugin");
 
 function EnvironmentPlugin(keys) {
-	this.keys = Array.isArray(keys) ? keys : Array.prototype.slice.call(arguments);
+	this.defaultValues = {};
+
+	if (Array.isArray(keys)) {
+		this.keys = keys;
+	}
+
+	else if (keys === Object(keys)) {
+		this.keys = Object.keys(keys);
+		this.defaultValues = keys;
+	}
+
+	else {
+		this.keys = Array.prototype.slice.call(arguments);
+	}
 }
-module.exports = EnvironmentPlugin;
 
 EnvironmentPlugin.prototype.apply = function(compiler) {
-	compiler.apply(new DefinePlugin(this.keys.reduce(function(definitions, key) {
-		var value = process.env[key];
+	var definitions = this.keys.reduce(function(defs, key) {
+		var value = process.env[key] || this.defaultValues[key];
 
 		if(value === undefined) {
 			compiler.plugin("this-compilation", function(compilation) {
-				var error = new Error(key + " environment variable is undefined.");
+				var error = new Error(
+					'EnvironmentPlugin - ' + key + ' environment variable is undefined. \033[0m\n\n' +
+					'You can pass an object with default values to suppress this warning. \n' +
+					'See http://webpack.github.io/docs/list-of-plugins.html#environmentplugin for example.'
+				);
+
 				error.name = "EnvVariableNotDefinedError";
 				compilation.warnings.push(error);
 			});
 		}
 
-		definitions["process.env." + key] = value ? JSON.stringify(value) : "undefined";
+		defs["process.env." + key] = value ? JSON.stringify(value) : "undefined";
```

Comment:
`value` could be false or empty string. `typeof value === "undefined ? JSON.stringify(value) : "undefined"`